core/templates/dev/head/App.ts

```diff
@@ -130,20 +130,23 @@ angular.module('oppia').config([
             return config;
           },
           responseError: function(rejection) {
+            var warningMessage = 'Error communicating with server.';
             // A rejection status of -1 seems to indicate (it's hard to find
             // documentation) that the response has not completed,
             // which can occur if the user navigates away from the page
             // while the response is pending, This should not be considered
             // an error.
             if (rejection.status !== -1) {
               $log.error(rejection.data);
-
-              var warningMessage = 'Error communicating with server.';
               if (rejection.data && rejection.data.error) {
                 warningMessage = rejection.data.error;
               }
               AlertsService.addWarning(warningMessage);
             }
+            var additionalLoggingInfo = warningMessage +
+              '\n URL: ' + rejection.config.url +
+              '\n data: ' + JSON.stringify(rejection.data);
+            $exceptionHandler(new Error(additionalLoggingInfo));
```

Comment:
The $exceptionHandler expects an Error object as a param to log stuff like line stacktrace and message (ie. exception.stack and exception.message).

I guess we could pass an object with an empty stacktrace and an appropriate error message. What do you think?

Reply:
Oh I see, I think I misread this. Let me rephrase. How about just doing `throw Error(...)` here, which is the way we normally invoke the exceptionHandler?

Reply:
The reason I am not throwing an error is because we want the return statement below to execute. We have unit tests for which check for such error cases and expect the return.

Reply:
I see, OK, that seems reasonable in this case. Perhaps add a detailed comment above this line, for future reference, explaining why we are adding this line of code and using $exceptionHandler directly instead of throwing an error.

(Also, in general, if a reviewer asks questions about something, this might be worth doing as a matter of course, because it indicates that at least one developer doesn't understand what is going on.)

Reply:
Noted, I'll leave a comment. Btw, I really like your tip! :)

Reply:
Thanks! I'll see if I can get that added to the "review process" page :)